src/Modules/OrchardCore.Commerce/Controllers/ShoppingCartController.cs

```diff
@@ -120,10 +120,22 @@ public async Task<ActionResult> Empty()
     public async Task<ActionResult> Update(ShoppingCartUpdateModel cart, string shoppingCartId)
     {
         var updatedLines = new List<ShoppingCartLineUpdateModel>();
-        foreach (var line in cart.Lines)
+
+        var lines = await cart.Lines.AwaitEachAsync(async line =>
+            (Line: line, Item: await _shoppingCartSerializer.ParseCartLineAsync(line)));
+
+        if (lines.Any(line => line.Item == null))
+        {
+            await _shoppingCartPersistence.StoreAsync(new ShoppingCart(), shoppingCartId);
```

Comment:
Do we need to instantiate the shopping cart each time?

Reply:
Each time? What do you mean?

Reply:
This line empties out the shopping cart by overwriting it with a new & empty instance. This is the safest way, given the current cart implementation. Creating a new `ShoppingCart` instance is cheap, because that type only contains two properties: a `string` and an empty `List`. Also this `if` block is error handling code that will only be called if the shopping cart was manipulated or corrupted, so I prioritize safety over performance.

Reply:
Looks good now